### ISSUE TITLE
Mypy doesn't catch certain type of errors when run inside tox

### ISSUE THREAD

**Issue by cburgdorf** [python/mypy#5771]
I'm facing a problem where `mypy` running inside `tox` does catch *certain* errors. It does catch these errors when invoked standalone though.

So while I'm not entirely sure if this error is caused by `tox` it isn't immediately obvious to me how `tox` could be responsible here.

```
$ tox -e py36-lint
py36-lint develop-inst-nodeps: /home/cburgdorf/Documents/hacking/ef/py-evm
py36-lint installed: asn1crypto==0.24.0,cffi==1.11.5,cryptography==2.3.1,cytoolz==0.9.0.1,eth-bloom==1.0.1,eth-hash==0.2.0,eth-keys==0.2.0b3,eth-typing==1.3.0,eth-utils==1.2.2,flake8==3.5.0,idna==2.7,lru-dict==1.1.6,mccabe==0.6.1,mypy==0.630,mypy-extensions==0.4.1,py-ecc==1.4.3,-e git+https://github.com/ethereum/py-evm.git@e7799c5fa400ef89c5a0134fc01273a15277842d#egg=py_evm,pycodestyle==2.3.1,pycparser==2.19,pyethash==0.1.27,pyflakes==1.6.0,rlp==1.0.3,six==1.11.0,toolz==0.9.0,trie==1.3.8,typed-ast==1.1.0
...
py36-lint runtests: commands[5] | mypy --follow-imports=silent --ignore-missing-imports --no-strict-optional --check-untyped-defs --disallow-incomplete-defs --disallow-untyped-defs --disallow-any-generics scripts/benchmark
py36-lint runtests: commands[6] | mypy --follow-imports=silent --ignore-missing-imports --no-strict-optional --check-untyped-defs --disallow-incomplete-defs --disallow-untyped-defs --disallow-any-generics -p p2p -p trinity
_____________________________________________________________________________________________________ summary _____________________________________________________________________________________________________
  py36-lint: commands succeeded
  congratulations :)
```

But then outside of `tox`
```
$ mypy --follow-imports=silent --ignore-missing-imports --no-strict-optional --check-untyped-defs --disallow-incomplete-defs --disallow-untyped-defs --disallow-any-generics -p p2p -p trinity
trinity/main.py:266: error: "ShutdownRequest" has no attribute "reason1"; maybe "reason"?
(venv) cburgdorf at craftbot in ~/Documents/hacking/ef/py-evm on christoph/mypy_weirdness
$ mypy --version
mypy 0.630

```

**Notice that this does not happen for all kind of errors but only certain ones**.

This has happened for other type errors, which passed our official CI but where caught locally (see: https://github.com/ethereum/py-evm/pull/1385)

**Steps to reproduce**

1. Clone https://github.com/cburgdorf/py-evm.git
2. `git checkout mypy_weirdness`
3. `pip install -e .[dev]`
4. `tox -e py36-lint (does not catch the error)`
5. `mypy --follow-imports=silent --ignore-missing-imports --no-strict-optional --check-untyped-defs --disallow-incomplete-defs --disallow-untyped-defs --disallow-any-generics -p p2p -p trinity`

**Comment by gvanrossum** [comment]
I'm sorry, you're on your own debugging this. This is undoubtedly some weird difference in your configuration. Good luck.

**Comment by cburgdorf** [comment]
@gvanrossum you're correct. Turns out our tox environment did not install the right dependencies for `mypy` to do it's job correctly. Closed by https://github.com/ethereum/py-evm/pull/1389
